### ISSUE TITLE
KA10: 340 CRT background color

### ISSUE THREAD

**Issue by larsbrinkhoff** [rcornwell/sims#150]
Here's my photo of a 340 CRT.  Maybe change the simulation background color to match?

CC @philbudne 
![20190325_143446](https://user-images.githubusercontent.com/775050/63159801-9ac91f00-c01c-11e9-9968-991db874bd9a.jpg)

**Comment by philbudne** [comment]
1. The color of the surface of the tube depends on room lighting,
	and/or any filter placed in front of the tube.

2. The same tube may have been used in the Type 30 display, so this
	isn't strictly PDP-10/340 related, and discussion belongs
	"upstream"

3. I find it hard to believe that this is the LEAST realistic
	part of the display simulation...

	I've long wondered if:

	 high intensity points "bloom" to appear
	 larger than low intensity points... whether this could be
	 simulated by intensifying neighboring display points, and more
	 recently have wondered whether 4K displays (with 3840?×?2160
	 pixels) might be able and whether current display libraries
	 might offer any portable abstractions to make aging easier...

	 The dynamic range of the real tubes was greater than modern
	 displays, and whether anything can be done to better approximate
	 that.

4. In many ways SimH gives an idealized user experience: Console
	devices were often Model 33 Teletypes, yet the default (AFAIK)
	is not to display only upper case, at 10 CPS, with the
	attendant noise, heat and smell (not to mention the effort
	needed to press any key).

	Likewise (again AFAIK), CPU execution, and disk/tape access
	speeds aren't as slow as real hardware.

	If/when SimH defaults to reproducing those aspects of the user
	experience, I think the default display surface color is a
	salient question (tho again, it depends on the room lighting).

	Until then, my preference is to default to black background, and
	perhaps have a SET command to allow it to be altered.

5. Finally I started with Doug Jones' VC8E simulation, but never
	"returned the favor" of adding VC8E support to SimH.

	Doug documents the interface here:
	http://homepage.divms.uiowa.edu/~jones/pdp8/man/vc8e.html

**Comment by rcornwell** [comment]
I agree, also these will appear much darker when the monitor is turned on.

What is your thoughts on adding a Triple III display for Waits. I have the basics done, just have not hooked it into the display.

**Comment by philbudne** [comment]
> ... what is your thoughts on adding a Triple III display for Waits.>

I know nothing about it, but http://xahlee.info/kbd/sail_keyboard.html
says it was a vector display, so the xy display code should be able to
support it.  For short persistence phosphor displays it's massive
overkill!

https://stacks.stanford.edu/file/druid:hb976hq8639/uuomanual.pdf (p.194)

Seems to suggest the display processor could drive 6 displays...

Something I wondered when I was doing the "glass tty" support for the
Bell Labs "Graphics II" processor for the PDP-7 was whether one could
do vector graphics in a browser window (perhaps a websocket connection
to transfer display lists, and rendering code in the browser), and be
able to render the button box and push buttons. And from there,
perhaps support front pannel lights and switches....  Mark P. was
unreceptive (to say the least)...

The Data Disc Display looks fun too!

I'm intrigued by https://www.saildart.org/j5/
the "L5" display has the text

"Test Pattern for SVG based III vector graphics terminal"

**Comment by larsbrinkhoff** [comment]
Regarding point 3, certainly this isn't the most pressing of all visual improvements that could be done.  BUT, it's very easy to change.  Low hanging fruit.

I do think there are blooming effects.  I also think using a GPU shader language would be a good fit for this.

About point 4, I have long maintained that it would be great to have a realistic teletype simulation.  I'm thinking physical model.  Heat and smell may be difficult.

I don't know about tape speed, I think Rich made it not too far off.... ;-)

**Comment by philbudne** [comment]
I randomly learned about HDR 4K video today: ie; 30bpp and 36bpp
(vs. the 24bpp we've used for decades).  So there's display hardware
capable of extended dynamic range!  Now support for "DeepColor" in the
entire software stack (sim_vid/SDL/window system) is another matter!
And if the increase in dynamic range makes it blatently obvious that
the aging system has a small number of color gradations, more fun will
ensue!

Looking at SDL docs I see SDL_SetPalette -- "Sets the colors in the
palette of an 8-bit surface" ... Even 15+ years ago when I started on
the XY simulator, I thought color maps had gone the way of the Dodo!
Playing color map games (or simply expressing the points on the screen
as a small integer aged intensity value) would simplify life.  EXCEPT
if you want to implement the rare tri-color 340-like display at AI!

Regarding using hardware shaders: Can it be done portably?

Can we agree that one person's "low hanging fruit" could be viewed as
gratuitous self gratification by another?

You _might_ get more respect if you came up with code that used the
in-screen webcam to determine ambient lighting, and color the
background appropriately, but I hope we can agree that it's overkill?!
[And I'll still leave my webcam closed (or taped over)!]

Personally, I wretch when I see vector scope simulations with a
background color other than black.
